extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -83,8 +93,12 @@ export class BindExpression {
     /** @const {string} */
     this.expressionString = expressionString;
 
-    /** @const {!./bind-expr-defines.AstNode} */
+    /** @const @private {!./bind-expr-defines.AstNode} */
     this.ast_ = parser.parse(this.expressionString);
+
+    /** @const @private {!Object} */
+    this.builtInScope_ = Object.create(null);
+    this.builtInScope_['Math'] = Math;
```

Comment:
I see your point, but this issue isn't limited to the `Math` object. A server-side template that performs `AMP.setState({{Username}}='foo')` will also overwrite `<amp-state>` data with the same `id` value. IMO this "gotcha" is worth the simplicity and flexibility it provides (e.g. `<amp-state>` can be used as default values for mutable data).

Personally not a fan of pipe syntax, but I can probably avoid adding `Math` to the scope as you mention by baking it in to the expression parser. Perhaps we should leave out `Math` altogether -- the expression `floor(-20)` shouldn't ever conflict with user data since we disallow custom functions.